src/actions/filesDropped/metadata.js

```diff
@@ -1,43 +1,53 @@
 import { rgb } from "d3-color";
 import { errorNotification, successNotification, warningNotification } from "../notifications";
 import { ADD_EXTRA_METADATA } from "../types";
-import { parseCsvTsv } from "./parseCsvTsv";
+import { parseCsv } from "./parseCsv";
 
 
 const handleMetadata = async (dispatch, getState, file) => {
   const fileName = file.name;
+  const reader = new FileReader();
+  reader.onload = async (event) => {
+    try {
+      const XLSX = (await import("xlsx/xlsx.mini")).default;
+      /* Convert tabular file to CSV string */
+      /* Only reads in the data from the first sheet of Excel files */
+      const workbook = XLSX.read(event.target.result, { type: 'binary' });
+      const firstSheet = workbook.Sheets[workbook.SheetNames[0]];
+      const sheetAsCsv = XLSX.utils.sheet_to_csv(firstSheet);
 
-  try {
-    /* Parse & interrogate the CSV file */
-    const {errors, data, meta} = await parseCsvTsv(file);
-    if (errors.length) {
-      console.error(errors);
-      throw new Error(errors.map((e) => e.message).join(", "));
+      /* Parse & interrogate the CSV file */
+      const {errors, data, meta} = await parseCsv(sheetAsCsv);
+      if (errors.length) {
+        console.error(errors);
+        throw new Error(errors.map((e) => e.message).join(", "));
+      }
+      const {coloringInfo, strainKey, latLongKeys, ignoredFields} = processHeader(meta.fields);
+      const rows = {};
+      data.forEach((d) => {rows[d[strainKey]]=d;});
+
+      /* For each coloring, extract values defined in each row etc */
+      const newNodeAttrs = {};
+      const newColorings = processColorings(newNodeAttrs, coloringInfo, rows, fileName); // modifies `newNodeAttrs`
+      const newGeoResolution = latLongKeys ? processLatLongs(newNodeAttrs, latLongKeys, rows, fileName) : undefined;
+      /* Fix errors in data & dispatch warnings here, as we cannot dispatch in the reducers */
+      const ok = checkDataForErrors(dispatch, getState, newNodeAttrs, newColorings, ignoredFields, fileName);
+      if (!ok) return undefined;
```

Comment:
This is not related to the PR, but I noticed that in this function some errors are manifested as exceptions (including the ones from libraries) and then dispatched to the store, but this one uses an error code and returns `undefined`.

Might be just different kind of errors, like ones we want to manifest vs ones to be silent. We might also bubble the exception handling up on the call stack, and decide how different errors are handled upstream.

Reply:
I see. Just noted that this particular bit looks inconsistent, I thought might be a bug in case this error gets erroneously silenced through return.

> Do you mean to hand these messages back up through custom errors that can be caught by the outer try/catch?

Yes, I usually try to do something like that. But probably not worth the trouble if already done otherwise.